Automattic/newspack-plugin
# sync `epic/ia` with `trunk`

### All Submissions:


### Changes proposed in this Pull Request:

Sync `epic/ia` with `trunk`. Changes (introduced in #3582) to Wizards needed to be migrated to new locations, some notable changes (13d6d55f6a3949ed17118a98369525cfc12ab442):
* Endpoint & callback for route `DELETE:/wizard/newspack-reader-revenue-wizard/reader-activation/emails/(?P<id>\d+)`  moved from `includes/wizards/class-reader-revenue-wizard.php` to `includes/wizards/audience/class-audience-donations.php`.
* Endpoint & callback for route `DELETE:/wizard/newspack-engagement-wizard/reader-activation/emails/(?P<id>\d+)` moved from `includes/wizards/class-engagement-wizard.php` to `includes/wizards/audience/class-audience-wizard.php`.
* Order notes checkbox added to `<BillingFields />` in `src/wizards/audience/components/billing-fields/index.tsx`
* "Checkout Configuration" group of components moved to Audience > Setup > Checkout & Payment.
* Audience > Setup tabs got wrapped in `WizardsSection`.
* Email reseting added to Transactional Emails & Newspack Settings > Emails.
* Recaptcha payload changes to accommodate `newspack_recaptcha_credentials ` option.

### How to test the changes in this Pull Request:

1. Checkout and build assets `git checkout feat/epic-ia-trunk-rasacc && npm run build`
2. Test email reseting:
   * Audience >  Setup > Transactional Emails
      * Navigate to _/wp-admin/admin.php?page=newspack-audience#/transactional-emails_
      * Pick an email and take note of the post id associated with it. This can be done by hovering over the "Edit" link.
      * Click "Reset", once request has completed, confirm the email has been reset (I did this by observing that the post id on the "Edit" link had changed).
![Screenshot 2024-12-04 at 14 49 31](https://github.com/user-attachments/assets/223cf422-7cba-436e-906f-e1f0aed744c6)
   * Newspack > Settings > Emails:
      * Navigate to _/wp-admin/admin.php?page=newspack-settings#/emails_
      * Observe "Cancellation" & "Welcome" emails have been added.
      * Follow the above repeatable steps starting at "Pick an email".
3. Navigate to _/wp-admin/admin.php?page=newspack-audience#/payment_
   * Scroll down to "Billing Fields".
       * Confirm newly added "Order notes" checkbox exists.
       * Perform editing & saving of checkbox and confirm changes persist between updates.
   * Scroll down to "Checkout Configuration"
       * Perform editing & saving of toggle and textareas and confirm changes persist between updates.
4. Navigate to _/wp-admin/admin.php?page=newspack-settings&scrollTo=newspack-settings-recaptcha#/_
   * Perform updates to recaptcha settings and confirm data persists between saves.

### Other information:
